tests/test_large_set.py

```diff
@@ -0,0 +1,152 @@
+docs_100M = get_test_docs(min_size=10 * 2**20, max_size=100 * 2**20)
+
+# Pytest parameter decorators
+for_each_10K_doc = pytest.mark.parametrize(
+    "doc", docs_10K, ids=[str(doc.name) for doc in docs_10K]
+)
+for_each_100K_doc = pytest.mark.parametrize(
+    "doc", docs_100K, ids=[str(doc.name) for doc in docs_100K]
+)
+for_each_10M_doc = pytest.mark.parametrize(
+    "doc", docs_10M, ids=[str(doc.name) for doc in docs_10M]
+)
+for_each_100M_doc = pytest.mark.parametrize(
+    "doc", docs_100M, ids=[str(doc.name) for doc in docs_100M]
+)
+
+
+@pytest.fixture
+def training(request: FixtureRequest) -> bool:
+    if request.config.getoption("--train"):
+        return True
+    else:
+        return False
+
+
+class TestLargeSet(TestCli):
+    def expected_container_output(self, input_file: Path) -> str:
+        # obtains the expected .log file
+        output_log_path = f"{input_file}.log"
+        with open(output_log_path, "r") as f:
+            return f.read()
+
+    def expected_success(self, input_file: Path) -> bool:
+        # obtains the expected result
+        expected_result_path = f"{input_file}.{CONTAINER_LOG_EXT}"
+        with open(expected_result_path, "r") as f:
+            last_line = f.readlines()[-1]  # result is in the last line
+            if "FAILURE" in last_line:
+                return False
+            elif "SUCCESS" in last_line:
+                return True
+            else:
+                raise ValueError(
+                    f"Container log file ({expected_result_path}) does not contain the result"
+                )
+
+    def run_doc_test(self, doc: Path, tmp_path: Path) -> None:
+        output_file_path = str(tmp_path / "output.pdf")
+        result = self.run_cli(
+            ["--output-filename", output_file_path, "--ocr-lang", "eng", str(doc)]
+        )
+        success = self.expected_success(doc)
+        if os.path.exists(output_file_path):
+            assert success, "Document was expected to fail but it didn't!"
+            result.assert_success()
+        else:
+            assert not success, "Document was expected to succeed but it didn't!"
+            result.assert_failure()
+
+    def train_doc_test(self, doc: Path, tmp_path: Path) -> None:
+        container_log_path = f"{doc}.{CONTAINER_LOG_EXT}"
+        if Path(container_log_path).exists():
+            os.remove(f"{doc}.{CONTAINER_LOG_EXT}")
+        output_file_path = str(tmp_path / "output.pdf")
+        self.run_cli(
+            ["--output-filename", output_file_path, "--ocr-lang", "eng", str(doc)],
+            env={"DZ_LOG_CONTAINER": "yes"},
+        )
+
+    def test_metadata(  # type: ignore[no-untyped-def]
+        self, record_testsuite_property
+    ) -> None:
+        # Add metadata about the test run to the test suite
+        record_testsuite_property(
+            "GIT_REF",
+            subprocess.check_output(["git", "rev-parse", "HEAD"]).decode()[:-1],
+        )
+        assert True
```

Comment:
I'm probably missing something, but where is this used?

Reply:
Turns out its reported in the final results, to know from which Git commit they were produced.